### PR TITLE
[dont merge] Hack to use the cpp asm-parser when parsing c++ objdumps

### PR DESCRIPTION
**Pull request by partouf**
Testing the PR thing

Todo:
- [ ] Refactor things

### CODE REVIEW

File: etc/scripts/objdump.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+OBJDUMP=/opt/compiler-explorer/gcc-10.1.0/bin/objdump
+ASMPARSER=/opt/compiler-explorer/asm-parser/build/src/asm-parser
```
Comment (mattgodbolt) [inline etc/scripts/objdump.sh:4]: Nice :)

File: lib/base-compiler.js [hunk 2]
```diff
@@ -174,7 +174,12 @@ export class BaseCompiler {
         const execOptions = {maxOutput: maxSize, customCwd: path.dirname(outputFilename)};
 
         const objResult = await this.exec(this.compiler.objdumper, args, execOptions);
-        result.asm = objResult.stdout;
+        if (objResult.stdout.startsWith('{')) {
```
Comment (mattgodbolt) [inline lib/base-compiler.js:177]: Ah: cunning - this seamlessly works with the old system?

File: lib/base-compiler.js [hunk 3]
```diff
@@ -941,10 +946,14 @@ export class BaseCompiler {
     }
 
     processAsm(result, filters, options) {
-        if ((options && options.includes('-emit-llvm')) || this.llvmIr.isLlvmIr(result.asm)) {
-            return this.llvmIr.process(result.asm, filters);
+        if (filters.binary && typeof result.asm !== 'string') {
```
Comment (mattgodbolt) [inline lib/base-compiler.js:949]: So; this says "if we're in binary mode and we already parsed it (cos it's not a string), then we're done...else do what we used to do. :+1:

**Review by mattgodbolt — APPROVED** [review]
Looks good to me :)

**Comment by partouf** [comment]
I hacked in a script that also replaced the arm gcc8.2 objdump, and this would fix #2107 as well. (because the stdin is parsed up until the "too many lines" bit and then the parser stops)

https://godbolt.org/beta/z/Wd9eKj

**Comment by partouf** [comment]
Obviously, this is not very effient -> https://github.com/compiler-explorer/compiler-explorer/pull/2270/commits/32d9f7645505fc2a6c564d7546633c4aa97cbdcb

So I think I have to introduce some new things like a customAsmParser property or something. I started doing that at first but it had some things making it difficult to introduce for just binary mode parsing. Will have another stab at the problem.

**Comment by mattgodbolt** [comment]
Excited to see all the progress on this :)

**Pull request closed without merging**